.github/workflows/trigger-dag-reuse-wf.yaml

```diff
@@ -34,17 +29,42 @@ on:  # yamllint disable-line rule:truthy
       astronomer_key_secret:
         description: 'astro cloud astronomer_key_secret'
         required: true
+      organization_id:
+        description: 'astro cloud organization_id'
+        required: true
+      bearer_token:
+        description: 'workspace bearer token'
+        required: true
 
 jobs:
   wait-for-deployment-to-be-ready-and-trigger-dag:
     runs-on: 'ubuntu-20.04'
     steps:
-      - name: Sleep and wait for astro cloud deployment
+
+      - name: Wait for deployment to be healthy
         run: |
-          echo "Current timestamp is" `date`
-          echo "Sleeping for ${{ inputs.wait_time }}"
-          echo "allowing the deployed image to be updated across all Airflow components."
-          sleep ${{ inputs.wait_time }}
+          astro_core_api="https://api.astronomer.io/v1alpha1/organizations/${{secrets.organization_id }}/\
+          deployments"
+          tries=15
+          health_flag=false
+
+          while [[ $tries -gt 0 && $health_flag == false ]]; do
+              sleep 120
```

Comment:
I think would be good to place this log at the end of while loop, just to see logs saying Deployment status is different than Health and it's waiting for deployment to be healthy in case if it becomes healthy in <2 mins.